### DISCUSSION TITLE
Unable to establish connection: CTS 602 and Waveshare Modbus TCP to RTU

### DISCUSSION THREAD

**Discussion by greg2011** [veista/nilan#147] [Q&A] [answered]
IP Address should be ok, device reachable.
I'm not sure about the destination IP, but used the one of my Raspberry with Homeassistant on docker. I've also tried my router IP, with no luck...

If I use an invalid IP or disconnect the waveshare, it tries to connect for about 3 sec. If the waveshare is there, the connection attempt lasts about 10sec, so there seems to be communication.
I looked for a manual or howto, I'm sure there's something else I'm missing?

I tried
`logger: default: warning logs: homeassistant.components.nilan: debug `
with no success, but couldn't find out the correct component name?

![image](https://github.com/user-attachments/assets/a93eaef3-f3f1-4a73-897b-0ed776b0b234)

![image](https://github.com/user-attachments/assets/6f14a9f6-48ae-433d-9e45-95fe8f7dda47)

**Comment by GruenerVeltliner** [comment]
Every IP-Adress in your system must be unique.
If your wave share device is in your local LAN it must have a separate IP which is unused in your router. 
With this IP you should can connect to your home assistant instance and Nilan integration.
Set a unique IP to wave share device and enter this IP to Nilan configuration.

**Reply by greg2011** [reply]
Hi, thanks - sorry if I wasn't clear.
(I'm aware of IP networking, subnet masks, octets, etc. and have set a static IP that's available, that's what I mean with address OK and reachable.)

Just not sure if it needs the destination IP for the traffic going to HA, or if this should be the DNS server. (Any of the two wouldn't work for me.)

**Comment by greg2011** [comment]
As a first step:
Can someone help me to activate the logging, please?

**Comment by greg2011** [comment, marked as answer]
I found the issue:
The work mode of the RS485 converter needs to be "TCP Server".

This is the correct configuration, working for me:
![image](https://github.com/user-attachments/assets/3012360b-ecad-4725-bc81-6ba52a867536)

**Reply by DUC750** [reply]
Thank you for sharing. Could you also mention which IPs you used for the Gateway and Destination IP? Unfortunately, it's not working with the settings for me yet :-(

**Reply by greg2011** [reply]
The Destination IP is my Raspberry with Home Assistant (I understand the "DNS" should refer to "DNS instead of IP address").
The Gateway is my Internet Router (but I don't think that it's required).

**Reply by DUC750** [reply]
Thank you. I've entered it just like that, but I still can't get a connection. It seems I might have another problem. Well, I guess I'll keep testing :-(

**Comment by Czdav1** [comment]
Hi,

I have a similar problem, I have a converter https://www.waveshare.com/rs485-to-eth-b.htm with PoE
But I don't want to connect with nilan VPL15 - CTS 602. I had the remote control connected to PIN 2,3 and GND. Set the converter the same (TCP server) but it won't connect and it won't light up blue (LINK DATA)

**Reply by Czdav1** [reply]
![image](https://github.com/user-attachments/assets/3e657160-fcdb-4132-9068-5adda8bdad33)

**Reply by Czdav1** [reply]
![image](https://github.com/user-attachments/assets/c0e0a290-2407-4a5c-8807-cf8d045149cb)


The other one is SOLAX and it works OK
